### PR TITLE
WT-11584 Fix test_checkpoint_stats test

### PR DESCRIPTION
**Pull request by jiechenbo**
The changes made in WT-10671 has a slight bug, where when we perform a retry of the loop we don't account for the extra checkpoint performed by the connection reopen.

### CODE REVIEW

File: test/suite/test_checkpoint04.py [hunk 1]
```diff
@@ -34,7 +34,7 @@
 from wtdataset import SimpleDataSet
 
 class test_checkpoint04(wttest.WiredTigerTestCase):
-    conn_config = 'cache_size=50MB,statistics=(all)'
+    conn_config = 'cache_size=50MB,statistics=(all,clear)'
```
Comment (etienneptl) [inline test/suite/test_checkpoint04.py:37]: Can you add a comment saying you are using `clear` to reset stats? You can use the previous comment that used to be there:
```
We don't want stats from earlier runs to interfere with later runs
```

**Review by etienneptl — APPROVED** [review]
LGTM modulo one comment

File: test/suite/test_checkpoint04.py (hunk 1 above)
Comment (jiechenbo) [reply test/suite/test_checkpoint04.py:38]: done! [2b6b66a](https://github.com/wiredtiger/wiredtiger/pull/9732/commits/2b6b66afcde3d290d674903790895a3298cfab81)

**Comment by jiechenbo** [comment]
evergreen merge

**Pull request merged**